main_ISTA.py

```diff
@@ -60,6 +60,8 @@ def __init__(self, data: Dataset, num_subsets: int = 7, step_size: float = 0.1,
                                                                     initial_image=data.OSEM_image)
         for f in obj_funs: # add prior to every objective function
             f.set_prior(data.prior)
+        data.prior.set_penalisation_factor(data.prior.get_penalisation_factor() / len(obj_funs))
+        data.prior.set_up(data.OSEM_image)
```

Comment:
also, what does this do and is it required?

Reply:
Generally, `set_up()` needs to be called for sirf.STIR objects before usage. It would do consistency checks, store look-up tables, whatever. Generally any `set` function could invalidate the `set_up`. Without checking, setting the penalisation factor doesn't invalidate the CudaRDP set_up, but conceptually, it's a dangerous thing to skip it. 

Ideally, STIR will check in the `set` function if it needs to set the object's `already_setup` member, but I suppose we might miss some. Also, ideally, calling `set_up` twice doesn't matter as the code could check if `set_up` is called with equivalent arguments. Without checking the Cuda RDP code, I'm not sure if that code is "ideal" in this sense...

Summary, I'd prefer to leave it in, unless we have a good reason to remove it...

Reply:
So I guess I'd reverse your latest commit.